pkg/liqoctl/offload/handler.go

```diff
@@ -72,6 +77,63 @@ func (o *Options) Run(ctx context.Context) error {
 	ctx, cancel := context.WithTimeout(ctx, o.Timeout)
 	defer cancel()
 
+	var offloadNamespaces []string
+	offloadNamespaces = append(offloadNamespaces, o.Namespaces...)
+
+	if o.LabelSelector != "" {
+		// Parse the label selector
+		selector, err := labels.Parse(o.LabelSelector)
+		if err != nil && o.LabelSelector != "" {
+			return fmt.Errorf("invalid label selector: %w", err)
+		}
+
+		// List namespaces
+		var selectedNsList corev1.NamespaceList
+		if err := o.CRClient.List(ctx, &selectedNsList, &client.ListOptions{LabelSelector: selector}); err != nil {
+			return fmt.Errorf("cannot list namespace objects: %w", err)
+		}
+
+		for _, ns := range selectedNsList.Items {
+			if !slices.Contains(offloadNamespaces, ns.Name) {
+				offloadNamespaces = append(offloadNamespaces, ns.Name)
+			}
+		}
+	}
+
+	if len(offloadNamespaces) == 0 {
+		o.Printer.Info.Println("No namespaces can be offloaded")
+		return nil
+	}
+
+	// Offload each namespace
+	var errors []error
+	for _, ns := range offloadNamespaces {
+		// Create a new options instance for each namespace to avoid state pollution
+		nsOptions := &Options{
+			Factory:                  o.Factory,
+			Namespace:                ns,
+			PodOffloadingStrategy:    o.PodOffloadingStrategy,
+			NamespaceMappingStrategy: o.NamespaceMappingStrategy,
+			RemoteNamespaceName:      o.RemoteNamespaceName,
+			ClusterSelector:          o.ClusterSelector,
+			OutputFormat:             o.OutputFormat,
+			Timeout:                  o.Timeout,
+		}
+
+		if err := nsOptions.runOffload(ctx); err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	if len(errors) > 0 {
+		return fmt.Errorf("encountered %d errors during offloading, error details: %v", len(errors), errors)
+	}
+
+	return nil
+}
+
+// runOffload encapsulates the core logic of the offload namespace command.
+func (o *Options) runOffload(ctx context.Context) error {
 	// Output the NamespaceOffloading resource, instead of applying it.
 	if o.OutputFormat != "" {
 		o.Printer.CheckErr(o.output())
```

Comment:
When the output is `yaml`, we need to append the end directive (`---`) to each of the printed object to produce a valid yaml output